src/main/java/org/folio/processing/mapping/mapper/writer/marc/MarcRecordModifier.java

```diff
@@ -794,4 +794,32 @@ private boolean isControlFieldsContains(List<ControlField> controlFields, Contro
         field.getTag().equals(controlField.getTag())
         && field.getData().equals(controlField.getData()));
   }
+
+  private boolean dataFieldsContain(List<DataField> dataFields, DataField dataField) {
+    return dataFields.stream().anyMatch(field -> fieldsEqual(field, dataField));
```

Comment:
Hi @RuslanLavrov. You can use compareTo method for checking equality between them.